tests/rest-api/test.py

```diff
@@ -0,0 +1,351 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+import io
+import os
+from pathlib import Path
+from random import randint
+from subprocess import Popen
+from time import sleep
+from typing import Generator
+from urllib.error import URLError
+from urllib.request import urlopen
+
+from gql import gql, Client
+from gql.transport.requests import RequestsHTTPTransport
+from pytest import Config, fixture, mark
+
+
+@fixture
+def client(pytestconfig: Config, request) -> Generator[Client, None, None]:
+    port = randint(20000, 65000)
+    root = Path(pytestconfig.getoption("root"))
+
+    env = {
+        "PYTHONPATH": str((root / "lib/python").resolve()),
+        "REVNG_ROOT": str(Path(root).resolve()),
+        "FLASK_RUN_PORT": str(port),
+        "FLASK_APP": "revng.restapi",
+        "FLASK_ENV": "development",
+    }
+    env = {**os.environ, **env}
+    out_fd = os.memfd_create("flask_debug", 0)
+    out = os.fdopen(out_fd, "w")
+
+    def print_server_log():
+        os.lseek(out_fd, 0, io.SEEK_SET)
+        out_read = os.fdopen(out_fd, "r")
+        print(out_read.read())
+
+    process = Popen(
+        ["/usr/bin/python3", "-m", "flask", "run"], env=env, stdout=out, stderr=out, text=True
+    )
+    for i in range(11):
+        if i == 10:
+            print_server_log()
+            raise ValueError()
+        try:
+            req = urlopen(f"http://127.0.0.1:{port}/", timeout=1.0)
+            if req.code == 200:
+                break
+            else:
+                sleep(1.0)
+        except (URLError, TimeoutError):
+            sleep(1.0)
```

Comment:
This makes me really sad, maybe isolated this in a function.

Reply:
Done